components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryRecoveryDataStore.java

```diff
@@ -48,6 +49,8 @@
     public void store(UserRecoveryDataDO recoveryDataDO) throws IdentityException {
         Registry registry = null;
         try {
+            String tenantDomain = IdentityTenantUtil.getTenantDomain(recoveryDataDO.getTenantId());
+            IdentityTenantUtil.initializeRegistry(recoveryDataDO.getTenantId(), tenantDomain);
```

Comment:
Shall we change the initializeRegistry method to have only one parameter? It is redundant to have both tenantId and domain as parameters.